### PR TITLE
Fill in test gaps for local-function-attributes

### PR DESCRIPTION
**Pull request by RikkiGibson**
Related to #38801

Some of the other scenarios like keyword recommending that I wanted to add support for in this PR are proving trickier than expected. So I will send that in a separate PR.

### CODE REVIEW

File: src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenLocalFunctionTests.cs [hunk 1]
```diff
@@ -5464,6 +5505,63 @@ static void validate(ModuleSymbol module)
...
+        }
+    }
+}
+";
+            CreateCompilation(source, parseOptions: TestOptions.RegularPreview).VerifyDiagnostics(
+                // (12,21): warning CS8321: The local function 'local1' is declared but never used
+                //         static void local1()
+                Diagnostic(ErrorCode.WRN_UnreferencedLocalFunction, "local1").WithArguments("local1").WithLocation(12, 21));
+
+            CreateCompilation(source, parseOptions: TestOptions.RegularPreview.WithPreprocessorSymbols("DEBUG")).VerifyDiagnostics();
```
Comment (RikkiGibson) [inline src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenLocalFunctionTests.cs]: I was slightly surprised by this behavior but I don't see a need to change it right now.

File: src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenLocalFunctionTests.cs [hunk 2]
```diff
@@ -5464,6 +5505,63 @@ static void validate(ModuleSymbol module)
...
+        {
+            Console.Write(""hello"");
+        }
+    }
+}
+";
+            CreateCompilation(source, parseOptions: TestOptions.RegularPreview).VerifyDiagnostics(
+                // (12,21): warning CS8321: The local function 'local1' is declared but never used
+                //         static void local1()
+                Diagnostic(ErrorCode.WRN_UnreferencedLocalFunction, "local1").WithArguments("local1").WithLocation(12, 21));
```
Comment (RikkiGibson) [inline src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenLocalFunctionTests.cs]: I was slightly surprised that the call didn't count as referencing the local function when the condition is not defined. It seems like given the restrictions around ConditionalAttribute there is basically no way to reference the local function if the condition for it is not defined.

**Comment by RikkiGibson** [comment]
@dotnet/roslyn-compiler FYI this a test-only change.

**Comment by RikkiGibson** [comment]
@dotnet/roslyn-compiler please review.

**Comment by RikkiGibson** [comment]
@jcouv I don't see your suggestion, did a comment get canceled by accident?

**Comment by jcouv** [comment]
Not sure what happened... Lost comment.
My suggestion was a test like `LookupInsideLocalFunctionAttribute` but without the local function, to simulate typing a local function with an attribute starting with the attribute. Not sure what happens in that case

**Comment by RikkiGibson** [comment]
Thanks, I will add a note to the test plan and follow up in a future PR.

**Pull request merged**